src/utils/updateTokenPrices.ts

```diff
@@ -103,3 +115,104 @@ async function getVeth2Price(
     return etherPrice
   }
 }
+
+export const getTokenPrice = (
+  tokens: BasicTokens,
+  dispatch: AppDispatch,
+  chainId: ChainId,
+): void => {
+  if (!tokens) return
+  const tokenAddresses = Object.keys(tokens)
+  const platform =
+    SUPPORTED_NETWORKS[chainId]?.coingeckoPlatformID || "ethereum"
+  const addressesChunk = chunk(
+    tokenAddresses,
+    MAX_ADDRESSES_PER_COINGECKO_REQUEST,
+  )
+  if (SUPPORTED_NETWORKS[chainId]) {
+    try {
+      void retry(
```

Comment:
Doing `retry(Promise.all)` rather than `Promise.all(retry)` may result in unnecessary fetches, since you will be retrying successful fetches when one fails.